### ISSUE TITLE
How to edit DATASETS: TRAIN&TEST in config.py to use the ADEChallengeData2016 dataset?

### ISSUE THREAD

**Issue by hiwmy0211** [KU-CVLAB/CAT-Seg#26]
hello, thank you for your excellent work!  I had some problems using the ADEChallengeData2016 dataset. I didn't know how to change the "TRAIN" and "TEST" parameters of "DATASETS:" in the config.py folder. I see coco_2017_train_stuff_all_sem_seg and coco_2017_test_stuff_all_sem_seg in the example, so if I want to use ADEChallengeData2016 dataset, What should I change these two parameters? thank you in advance for your reply.

**Comment by hsshin98** [comment]
Hi, and thanks for having interest in our work!
This would require to register the training set of ADE20k into detectron2, which could be done in: https://github.com/KU-CVLAB/CAT-Seg/blob/6d3a188af95165147fe2f34a8237fa7d2633e784/cat_seg/data/datasets/register_ade20k_150.py#L19
First, you should preprocess the annotation files into detectron2 format(cat_seg/data/datasets/register_ade20k_150.py) for the training set too, because currently only prepare the validation set.
After the processing, dding a line with the name and the directory to the training set should do the work, for example:
("train", "images/training", "annotations_detectron2/validation")
We follow commonly used dataloader configuration for detectron2, so having a look at other works(https://github.com/facebookresearch/MaskFormer/tree/main) with detectron might also help.
Sorry for the late reply, and let me know if you have further problems!
